### PR TITLE
Adds dependencies cron job to docker-compose files

### LINKED ISSUE #75: Think about scheduled task
Both the spark job and any mysql backend need to have a periodic task run.  The spark job needs to be run in order to produce the dependency graph. The mysql backend needs to "delete from" in order to implement TTL. There's probably a simple pattern in docker nowadays we can use to implement both.

See https://github.com/openzipkin/zipkin/issues/792#issuecomment-160589617 from @MatMoore about needing TTL for MySQL
Also @prat0318 was asking about freshening the dependencies data from https://github.com/openzipkin/zipkin-dependencies-spark

cc @yurishkuro @jcarres-mdsol

### LINKED ISSUE #76: Add image for zipkin-dependencies-spark
zipkin-dependencies-spark should be a daily scheduled task (#75)

https://bintray.com/openzipkin/maven/zipkin-dependencies-spark/1.25.3/view#files

### PR DESCRIPTION
**Pull request by adriancole**
This adds another service which runs the [dependencies job](https://github.com/openzipkin/docker-zipkin-dependencies) via cron.

Fixes #75
Fixes #76

### CODE REVIEW

**Comment by adriancole** [comment]
@prat0318 @jcarres-mdsol do you have any opinions on how the dependencies job ends up in docker? right now, it is an admin container

File: dependencies/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+FROM openzipkin/jre-full:1.8.0_72
+MAINTAINER OpenZipkin "http://zipkin.io/"
+
+ARG STORAGE_TYPE
+ENV ZIPKIN_REPO https://jcenter.bintray.com
+ENV DEPENDENCIES_VERSION 1.42.1
```
Comment (adriancole) [inline dependencies/Dockerfile]: this will change

File: dependencies/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Use to set heap, trust store or other system properties.
+ENV JAVA_OPTS -Djava.security.egd=file:/dev/./urandom
+
+WORKDIR /zipkin-dependencies
+
+RUN curl -SL $ZIPKIN_REPO/io/zipkin/zipkin-dependencies-spark-$STORAGE_TYPE/$DEPENDENCIES_VERSION/zipkin-dependencies-spark-$STORAGE_TYPE-$DEPENDENCIES_VERSION-all.jar > zipkin-dependencies.jar
```
Comment (adriancole) [inline dependencies/Dockerfile]: this url will also change per https://github.com/openzipkin/zipkin-dependencies-spark/issues/31

**Comment by jcarres-mdsol** [comment]
So admin tasks is a way to have Dockerfiles that can run optionally with dependencies but without hard dependencies on all the images?

Sounds good. Probably will be difficult to make it work in ECS but we'll figure out something

**Comment by adriancole** [comment]
well, it is more about docker-compose than the dockerfile. If not using
docker-compose, you'd just run the image every day somehow.. Not sure if
run-task is the answer or not
http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_run_task.html

**Comment by adriancole** [comment]
simplifying the build by making a single spark binary https://github.com/openzipkin/zipkin-dependencies/pull/38

**Comment by adriancole** [comment]
I think the docker config could be best like a cron where it processes the day before just after midnight, and the current day on the hour. The latter will keep the "today" data fresh eventhough it will do some reprocessing.

**Comment by adriancole** [comment]
alright, the actual image is hosted here, version 1.0 (so watch for updates)

https://github.com/openzipkin/docker-zipkin-dependencies

This change simply adds it to the default docker-compose file, specifically using the bundled cron config

**Comment by adriancole** [comment]
booya... works!

Here's an example of elasticsearch job automagically doing dependencies at the top of the hour
```
elasticsearch   | [2016-07-22 10:58:28,834][INFO ][cluster.metadata         ] [Gorr] [zipkin-2016-07-22] update_mapping [span]
dependencies    | crond: USER root pid   5 cmd run-parts /etc/periodic/15min
dependencies    | crond: USER root pid   6 cmd run-parts /etc/periodic/hourly
dependencies    | Processing spans from zipkin-2016-07-22/span
dependencies    | 11:00:01.569 [main] WARN  o.a.hadoop.util.NativeCodeLoader - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
dependencies    | Saving dependency links to zipkin-2016-07-22/dependencylink
elasticsearch   | [2016-07-22 11:00:05,121][INFO ][cluster.metadata         ] [Gorr] [zipkin-2016-07-22] create_mapping [dependencylink]
dependencies    | Dependencies: [{parent=brave-resteasy-example, child=brave-resteasy-example, id=brave-resteasy-example|brave-resteasy-example, callCount=1}]
```

**Pull request merged**